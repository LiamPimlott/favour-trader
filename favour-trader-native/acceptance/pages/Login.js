'use strict';

let I;

module.exports = {

    _init() {
        I = require('../../steps_file.js')();
    },

    login(email, password) {
        I.fillField('~Email-Address', email);
        I.fillField('~Password', password );
        I.tap('~Login');
    },

    goToSignup() {
        I.tap('Sign Up')
    }

}
