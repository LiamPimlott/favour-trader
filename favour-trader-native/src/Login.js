import React, { Component } from 'react';
import { StyleSheet, View, Text, TextInput, TouchableOpacity, KeyboardAvoidingView, Image } from 'react-native';
import { StackNavigator, NavigationActions } from 'react-navigation';

export default class Login extends React.Component {
	static navigationOptions = { 
		header: null
	}
	render(){
		const { navigate } = this.props.navigation;
		
		return( 

			<KeyboardAvoidingView behavior="padding" style={styles.container}>
				<Image source={require('../images/test.jpg')} style={styles.backgroundImage}></Image>
				<View style={styles.container}>   
						<Text style={styles.title} >Welcome to Favour Trader</Text>
						<Text style={styles.intro}>Login to start trading!</Text> 
					<View>
						<TextInput
							placeholder="username"
							style={styles.input}
							onSubmitEditing={() => this.passwordInput.focus()}
						/>
						
						<TextInput 
							placeholder="password" 
							style={styles.input}
							secureTextEntry
							ref={(input) => this.passwordInput = input}
						/>
						 
						<TouchableOpacity 
							style={styles.buttonContainer}
							onPress = { () => navigate('Home') }    
						>
							<Text style={styles.buttonText}>LOGIN</Text> 
						</TouchableOpacity> 
						
					
					</View> 
				</View> 
					<TouchableOpacity 
						style={styles.buttonSignUp}
						onPress = { () => navigate('SignUp') }
					>
						<Text style={styles.buttonTextAcc}>No Account?</Text>
						<Text style={styles.buttonText}>SIGN UP</Text>
					</TouchableOpacity>
		  </KeyboardAvoidingView>
		  
		);
	}
}

const styles = StyleSheet.create({
	container: {  
		flex: 1,
		backgroundColor: 'transparent',
		alignItems: 'center',
		justifyContent: 'center',
	},
	backgroundImage: {
		position: 'absolute',
		bottom: 0        
	},
	input: { 
		backgroundColor: '#9978C2',
		padding: 20,
		marginBottom: 5,      
		fontSize: 20
	},
	buttonContainer: { 
		backgroundColor: '#6F4B9B',
		justifyContent: 'center',
		height: '20%',
		width: 400,
		marginTop: 10, 
		marginBottom: 5,
		borderRadius: 20
	},
	buttonSignUp: { 
		backgroundColor: 'transparent', 
		justifyContent: 'center', 
		height: 75,
		width: 200,
		borderRadius: 25 
	},
	buttonTextAcc: {
		textAlign: 'center',
		fontSize: 15
	},	
	buttonText: {
		textAlign: 'center',
		fontSize: 25 
	},
	title: {  
	  fontWeight: 'bold',  
	  fontSize: 45 
	}, 
	intro: { 
	  fontSize: 20 
	}
});