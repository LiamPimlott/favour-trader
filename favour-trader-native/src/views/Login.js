import React, { Component } from 'react';
import { StyleSheet, View, Text, TextInput, TouchableOpacity, KeyboardAvoidingView, Alert, Image } from 'react-native';
import AuthService from "../components/AuthService";
import { Input, Icon, Button, Card, Divider } from 'react-native-elements';

export default class Login extends React.Component {
    constructor(){
        super();
        this.state = {
            failedAttempt: false,
            email: '',
            password: '',
        };
        this.authService = new AuthService();
        this.submit = this.submit.bind(this);
    }

    submit() {
        const { navigate } = this.props.navigation;
        this.authService.login(this.state.email, this.state.password)
            .then(res => {
                if (res.success && res.token) {
                    navigate('Home')
                } else {
                    this.setState({
                        failedAttempt: true,
                    });
                    if(this.state.failedAttempt){
                        this.renderErrorText(res.message);
                    }
                }
            })
            .catch(err => {
                Alert.alert('Exception', err.message);
            });
    }

    renderErrorText(message) {
        Alert.alert('Invalid Login', message);
    }

    render(){
        const { navigate } = this.props.navigation;
        return(
            <KeyboardAvoidingView behavior="padding" style={styles.container}>
                <Card
                    title={'Welcome to Favour â‡‹ Trader'}>
                    <Text style={styles.intro}>Login to start trading!</Text>
                    <View style={styles.input}>
                        <Input
                            placeholder='E-mail Address'
                            leftIcon={
                                <Icon
                                    name='mail-outline' />
                            }
                            autoCapitalize={'none'}
                            onChangeText={(text) => this.setState({email: text})}
                        />
                    </View>
                    <View style={styles.input}>
                        <Input
                            placeholder='Password'
                            leftIcon={
                                <Icon
                                    name='lock-outline' />
                            }
                            autoCapitalize={'none'}
                            secureTextEntry={true}
                            onChangeText={(text) => this.setState({password: text})}
                        />
                    </View>

                    <Button
                        backgroundColor='#03A9F4'
                        buttonStyle={styles.button}
                        title='Login'
                        onPress = { this.submit} />

                    <Divider style={styles.divider}/>
                    <View style={styles.footerView}>
                        <Text style={{textAlign: 'center'}}>No account?</Text>
                        <Button
                            clear={true}
                            titleStyle={{ color: "#03A9F4" }}
                            title='Sign Up'
                            onPress = { () => navigate('Signup') } />
                    </View>
                </Card>
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'transparent',
        alignItems: 'center',
        justifyContent: 'center',
    },
    input: {
        marginTop: 10,
        marginBottom: 15,
    },
    button: {
        width: 300,
        borderRadius: 0,
        marginLeft: 0,
        marginRight: 0,
        marginBottom: 10
    },
    intro: {
        fontSize: 15,
        textAlign: 'center',
        marginBottom: 10
    },
    divider: {
        width: 300,
        backgroundColor: 'grey',
        marginTop: 15,
    },
    footerView: {
        marginTop: 20,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    }
});