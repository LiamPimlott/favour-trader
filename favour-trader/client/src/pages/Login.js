import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import {Card,  Form, Icon, Input, Button } from 'antd';
import './Login.css';
import {message} from "antd/lib/index";
const FormItem = Form.Item;

class Login extends Component {
    constructor() {
        super();
        this.state = {
            email: "",
            password: "",
            redirect: false,
            failedAttempt: false,
            iconLoading: false,
        };

        this.renderErrorText = this.renderErrorText.bind(this);
        this.submit = this.submit.bind(this);
    }

    componentWillMount() {
        const { authService } = this.props;
        if (authService.loggedIn()) {
            this.setState({
                redirect: true,
            });
        }
    }

    submit() {
        const {authService} = this.props;
        this.props.form.validateFields((err, values) => {
            if (!err) {
                authService.login(
                    values.emailAddress,
                    values.password,
                    )
                    .then(res => {
                        if (res.success && res.token) {
                            this.setState({
                                redirect: true,
                            });
                        } else {
                            this.setState({
                                failedAttempt: true,
                                iconLoading: true
                            });
                            if(this.state.failedAttempt){
                                this.renderErrorText(res.message);
                            }
                        }
                    })
                    .catch(err => {
                        alert(err);
                    });
            }
        })

    }

    renderErrorText(err) {
        return (
            message.error(err, 2 , ()=> {
                this.setState({
                    iconLoading: false
                });
            })
        );
    }

    render() {
        const { getFieldDecorator } = this.props.form;
        const title = <h5>Favor <Icon type={'swap'}/> Trader</h5>;
        return (
            <div className={'container'}>
                {
                    (this.state.redirect) ? (<Redirect to={'/'}/>) : ('')
                }
                <Card title={title} bordered={true} className={'login-form-card'}>
                    <Form className="login-form">
                        <FormItem>
                            {
                                getFieldDecorator('emailAddress', {
                                    rules: [{ required: true, message: 'Please input your E-mail address!' },
                                            {type: 'email', message: 'The input is not valid E-mail!'}],
                                    validateTrigger: 'onBlur'
                                })(
                                <Input
                                    prefix={<Icon type="mail" style={{ color: 'rgba(0,0,0,.25)' }} />}
                                    placeholder="Email Address"
                                    type="email"
                                    name="email"
                                    id="email" />
                                )
                            }
                        </FormItem>
                        <FormItem>
                            {
                                getFieldDecorator('password', {
                                    rules: [{ required: true, message: 'Please input your password!' }],
                                    validateTrigger: 'onBlur',
                                })(
                                    <Input type="password"
                                           name="password"
                                           id="password"
                                           placeholder=" Password"
                                           prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}/>
                                )
                            }
                        </FormItem>
                        <FormItem>
                            <Button type="primary" htmlType="submit"
                                    onClick={this.submit} loading={this.state.iconLoading}  className="login-form-button">
                                Log in
                            </Button>
                            Or <a href="/create-account">register now!</a>
                        </FormItem>
                    </Form>
                </Card>
            </div>
        );
    }
}

const WrappedLoginForm = Form.create()(Login);
export default WrappedLoginForm;