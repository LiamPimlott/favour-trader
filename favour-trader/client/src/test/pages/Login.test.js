import React from 'react';
import renderer from 'react-test-renderer';
import moxios from 'moxios';
import { mount, configure } from 'enzyme';
import Login from '../../pages/Login.js';
import AuthServiceMock from '../components/AuthServiceMock.js';
import Adapter from 'enzyme-adapter-react-16';

var authServiceMock, wrapper;
//formElements
var emailInput, passwordInput, submitBtn;

beforeEach(function () {
    configure({ adapter: new Adapter() });
    moxios.install()
    authServiceMock = new AuthServiceMock();
    wrapper = mount(<Login authService={authServiceMock} />);
    //formElements
    emailInput = wrapper.find('input#emailAddress');
    passwordInput = wrapper.find('input#password');
    submitBtn = wrapper.find('Button[type="submit"]');
})

afterEach(function () {
    moxios.uninstall()
    authServiceMock = null;
    wrapper = null;
    emailInput = null;
    passwordInput = null;
})

it('renders correctly', () => {
    var authServiceMock = new AuthServiceMock();
    const tree = renderer
        .create(<Login authService={authServiceMock}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('verify Login Page', function () {
    let cardTitle = wrapper.find('h5');
    expect(cardTitle.length).toEqual(1);
    expect(wrapper.contains([<a href="/create-account">register now!</a>])).toEqual(true);
    expect(wrapper.contains([<span>Log in</span>])).toEqual(true);
    expect(emailInput.length).toEqual(1);
    expect(passwordInput.length).toEqual(1);
});

it('should verify error validation messages on Empty fields', function () {
    emailInput.simulate('blur');
    passwordInput.simulate('blur');
    expect(wrapper.find('.ant-form-explain').length).toBe(2);
    expect(wrapper.find('.ant-form-explain').get(0).props.children).toBe('Please input your E-mail address!');
    expect(wrapper.find('.ant-form-explain').get(1).props.children).toBe('Please input your password!');
});

it('should verify error validation messages on invalidate fields', function () {
    emailInput.simulate('change', {target: {name: 'email', value: 'i'}});
    emailInput.simulate('blur');

    expect(wrapper.find('.ant-form-explain').length).toBe(1);
    expect(wrapper.find('.ant-form-explain').get(0).props.children).toBe("The input is not valid E-mail!");
});

