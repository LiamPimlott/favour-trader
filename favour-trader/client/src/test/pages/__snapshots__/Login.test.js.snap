// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  
  <div
    className="ant-card login-form-card ant-card-bordered"
  >
    <div
      className="ant-card-head"
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          <h5>
            Favor 
            <i
              className="anticon anticon-swap"
            />
             Trader
          </h5>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={undefined}
    >
      <form
        className="ant-form ant-form-horizontal login-form"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={undefined}
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span
                  className="ant-input-affix-wrapper"
                  style={undefined}
                >
                  <span
                    className="ant-input-prefix"
                  >
                    <i
                      className="anticon anticon-mail"
                      style={
                        Object {
                          "color": "rgba(0,0,0,.25)",
                        }
                      }
                    />
                  </span>
                  <input
                    className="ant-input"
                    data-__field={
                      Object {
                        "name": "emailAddress",
                      }
                    }
                    data-__meta={
                      Object {
                        "name": "emailAddress",
                        "originalProps": Object {
                          "disabled": false,
                          "id": "email",
                          "name": "email",
                          "placeholder": "Email Address",
                          "prefix": <Icon
                            style={
                                  Object {
                                        "color": "rgba(0,0,0,.25)",
                                      }
                            }
                            type="mail"
                      />,
                          "prefixCls": "ant-input",
                          "type": "email",
                        },
                        "ref": null,
                        "rules": Array [
                          Object {
                            "message": "Please input your E-mail address!",
                            "required": true,
                          },
                          Object {
                            "message": "The input is not valid E-mail!",
                            "type": "email",
                          },
                        ],
                        "trigger": "onChange",
                        "validate": Array [
                          Object {
                            "rules": Array [
                              Object {
                                "message": "Please input your E-mail address!",
                                "required": true,
                              },
                              Object {
                                "message": "The input is not valid E-mail!",
                                "type": "email",
                              },
                            ],
                            "trigger": Array [
                              "onBlur",
                            ],
                          },
                        ],
                        "validateTrigger": "onBlur",
                        "valuePropName": "value",
                      }
                    }
                    disabled={false}
                    id="emailAddress"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Email Address"
                    style={null}
                    type="email"
                    value=""
                  />
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={undefined}
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span
                  className="ant-input-affix-wrapper"
                  style={undefined}
                >
                  <span
                    className="ant-input-prefix"
                  >
                    <i
                      className="anticon anticon-lock"
                      style={
                        Object {
                          "color": "rgba(0,0,0,.25)",
                        }
                      }
                    />
                  </span>
                  <input
                    className="ant-input"
                    data-__field={
                      Object {
                        "name": "password",
                      }
                    }
                    data-__meta={
                      Object {
                        "name": "password",
                        "originalProps": Object {
                          "disabled": false,
                          "id": "password",
                          "name": "password",
                          "placeholder": " Password",
                          "prefix": <Icon
                            style={
                                  Object {
                                        "color": "rgba(0,0,0,.25)",
                                      }
                            }
                            type="lock"
                      />,
                          "prefixCls": "ant-input",
                          "type": "password",
                        },
                        "ref": null,
                        "rules": Array [
                          Object {
                            "message": "Please input your password!",
                            "required": true,
                          },
                        ],
                        "trigger": "onChange",
                        "validate": Array [
                          Object {
                            "rules": Array [
                              Object {
                                "message": "Please input your password!",
                                "required": true,
                              },
                            ],
                            "trigger": Array [
                              "onBlur",
                            ],
                          },
                        ],
                        "validateTrigger": "onBlur",
                        "valuePropName": "value",
                      }
                    }
                    disabled={false}
                    id="password"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder=" Password"
                    style={null}
                    type="password"
                    value=""
                  />
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={undefined}
        >
          <div
            className="ant-form-item-control-wrapper"
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <button
                  className="ant-btn login-form-button ant-btn-primary"
                  onClick={[Function]}
                  type="submit"
                >
                  <span>
                    Log in
                  </span>
                </button>
                Or 
                <a
                  href="/create-account"
                >
                  register now!
                </a>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
